### PR TITLE
Shereen purchase equipment

### PR DESCRIPTION
**Pull request by pshereen**
# Description
Adds functionality for the user to place purchase order for equipment. [/bmdashboard/equipment/purchase/] (http://localhost:3000/bmdashboard/equipment/purchase/).

Implements # (WBS) 9.2.2

Screenshot 2024-05-09 105828

## Related PRS (if any):
This backend PR is related to the #2251 front-end PR.
To test this backend PR you need to checkout https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2251 frontend PR.
…

## Main changes explained:
- Added controller function for equipment Purchase.
- Added Routing for purchase equipment request
…

## How to test:
1. check into current branch.
2. Run npm install and npm run build && npm start to run the backend locally.
3. Clear site data/cache
4. Log in as any user
5. Go to /bmdashboard (http://localhost:3000/bmdashboard) and log in again.
6. Navigate to [/bmdashboard/equipment/purchase] (http://localhost:3000/bmdashboard/equipment/purchase/)
7. Place a purchase order for an equipment.
8. Verify that the green success message has been displayed after the form submission.

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HGNRest/assets/7041610/7f3fd264-1599-4385-85c5-995b541c0344

### CODE REVIEW

**Review by Sandhya1236 — APPROVED** [review]
I have tested the changes successfully by checking out the current branch, running npm install, npm run build, and npm start to set up the backend locally. After clearing site data and cache, I logged in as a user and accessed the BM Dashboard at http://localhost:3000/bmdashboard. I then navigated to the Equipment Purchase page at http://localhost:3000/bmdashboard/equipment/purchase and placed a purchase order for equipment. Upon form submission, the green success message was displayed, confirming the successful placement of the order. All functionalities worked as expected, and the feature appears stable. The PR is ready for merging. Please have a look at the video of my testing for reference.


https://github.com/OneCommunityGlobal/HGNRest/assets/79351059/928d8187-c4da-423a-b142-2c7b7ffebd40

**Review by peter6866 — APPROVED** [review]
The backend implementation for equipment purchase orders looks solid! The new endpoint for handling purchase order submissions is well-structured and integrates smoothly with the frontend. 
I also left a detailed review for the frontend PR.

**Review by jinxyou — APPROVED** [review]
I followed the steps given in the instructions, and the purchase seems to be working fine. Great work!
Screenshot 2024-06-15 195357

File: requirements/badgeController/deleteBadge.md
Comment (ptrpengdev) [inline requirements/badgeController/deleteBadge.md:1]: I noticed multiple MD files were deleted from the requirements folder. Is this expected behavior?

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**